<launch>
	
	<arg name="loop_rate"/>
	<arg name="record_bag"/>
	
	<node
		pkg="beginner_tutorials" 
		type="string_flip" 
		name="string_flip"
		required="true"
	/>
	<node 
		pkg="beginner_tutorials" 
		type="listener" 
		name="listener"
		launch-prefix="xterm -e"
	/>
	<node 
		pkg="beginner_tutorials" 
		type="talker" 
		name="talker"
		launch-prefix="xterm -e">
		<param name="loopRate" value="$(arg loop_rate)" type="double"/>
	</node>
	<node
		pkg="rosbag"
		type="record"
		name="tutorial_bag"
		args="-a --duration=20 -O tutorialWeek11.bag"
		if="$(arg record_bag)"
	/>

</launch>
